### PR TITLE
Fix crash in punch and roll recording

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #5080

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp [hunk 1]
```diff
@@ -988,7 +988,7 @@ void WaveformView::DoDraw(TrackPanelDrawingContext &context, size_t channel,
    channel = (track.IsLeader() ? 0 : 1);
 
    // TODO wide wave tracks -- remove this workaround
-   auto pLeader = *track.GetOwner()->Find(&track);
```
Comment (saintmatthieu) [inline src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp:991]: Is this the line that was crashing? To my wobbly understanding returns `GetOwner()` and `GetHolder()` a pointer to the same object, only the former a the lock of a weak pointer and the latter, a raw pointer. But that wouldn't explain why this change would fix the crash.

Comment (Paul-Licameli) [reply src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp:991]: There is some complication during recording.  RegisterPendungChangedTrack sets up one or two tracks for which owner and holder differ.

**Comment by saintmatthieu** [comment]
Trying to approve (where is that button?)

**Review by saintmatthieu — APPROVED** [review]

**Comment by dozzzzer** [comment]
The PR partially fixes the issue #5080 
However, there's still a problem when recording in the Waveform view initially and then resuming recording in the Spectrogram view; it is logged separately as #5094
